File: src/MessagesService/MessagesService.Core/Models/Notification.cs
using MessagesService.Core.Enums;

namespace MessagesService.Core.Models
{
    public class Notification
    {
        public string Id { get; set; }
        public Guid RecipientId { get; set; }
        public string Content { get; set; }
        public NotificationType Type { get; set; }
        public NotificationStatus Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public Dictionary<string, string> Metadata { get; set; }
    }
}

File: frontend/src/app/core/models/notification.ts
import { NotificationStatus } from '../enums/notificationStatus';
import { NotificationType } from '../enums/notificationType';

export default interface Notification {
  id: string;
  recipientId: string;
  content: string;
  type: NotificationType;
  status: NotificationStatus;
  createdAt: Date;
  metadata: { [key: string]: string };
}
